chainsaw-demo/src/logging.rs

```diff
@@ -60,3 +64,14 @@ pub fn set_global_logger(subscriber: impl Subscriber + Send + Sync) -> Result<()
 
     Ok(())
 }
+
+pub fn http_trace_layer() -> TraceLayer<SharedClassifier<ServerErrorsAsFailures>> {
+    TraceLayer::new_for_http()
+        .make_span_with(
+            DefaultMakeSpan::new()
+                .include_headers(true)
```

Comment:
This isn't quite what we'd need, we should instead:
- Read x-b3-trace-id and x-b3-spanid headers.
- Create a new span using those exact values.
- Expose those values such that they appear in structured logging.

But happy for that to be figured out later.